### PR TITLE
Context replacment

### PR DESCRIPTION
**Pull request by orwa97**

### CODE REVIEW

File: src/pages/Main/muqtatif/Muqtatif.jsx [hunk 1]
```diff
@@ -1,19 +1,25 @@
...
-import { SketchPicker } from "react-color";
-import Settings from "./muqHeader/settings/Settings";
 import MuqHeader from "./muqHeader/MuqHeader";
-
+import { useLocation } from "react-router-dom";
+import queryString from "query-string";
+import { defaultValues } from "./muqHeader/settings/defaultValues";
 const Muqtatif = (props) => {
+  const location = useLocation();
+  const params = queryString.parse(location.hash);
```
Comment (Hartha-aloufi) [inline src/pages/Main/muqtatif/Muqtatif.jsx]: put this in useMemo

File: src/pages/Main/muqtatif/Muqtatif.jsx [hunk 2]
```diff
@@ -1,19 +1,25 @@
...
+  const settingsValues = Object.fromEntries(
+    Object.entries(params).map(([k, v], i) => [k, v.replace(/['"]+/g, "")])
+  );
+
+  console.log(settingsValues.fontSize);
```
Comment (Hartha-aloufi) [inline src/pages/Main/muqtatif/Muqtatif.jsx]: remove this

File: src/pages/Main/muqtatif/Muqtatif.jsx [hunk 3]
```diff
@@ -1,19 +1,25 @@
...
+  const [fontSize, setFontSize] = useState(defaultValues.fontSize);
```
Comment (Hartha-aloufi) [inline src/pages/Main/muqtatif/Muqtatif.jsx]: move this up. put all useState in the top

File: src/pages/Main/muqtatif/Muqtatif.jsx [hunk 4]
```diff
@@ -1,19 +1,25 @@
...
+  useEffect(() => {
+    setFontSize(settingsValues.fontSize || defaultValues.fontSize);
+    console.log(fontSize);
```
Comment (Hartha-aloufi) [inline src/pages/Main/muqtatif/Muqtatif.jsx]: remove this

File: src/pages/Main/muqtatif/Muqtatif.jsx [hunk 5]
```diff
@@ -1,19 +1,25 @@
...
+  const location = useLocation();
+  const params = queryString.parse(location.hash);
+  const settingsValues = Object.fromEntries(
+    Object.entries(params).map(([k, v], i) => [k, v.replace(/['"]+/g, "")])
+  );
+
+  console.log(settingsValues.fontSize);
+  const [fontSize, setFontSize] = useState(defaultValues.fontSize);
+  useEffect(() => {
+    setFontSize(settingsValues.fontSize || defaultValues.fontSize);
```
Comment (Hartha-aloufi) [inline src/pages/Main/muqtatif/Muqtatif.jsx]: convert this to useMemo and calculate `settingsValues` here

File: src/pages/Main/muqtatif/Muqtatif.jsx [hunk 6]
```diff
@@ -107,129 +58,25 @@ const Muqtatif = (props) => {
...
-            width={textBgWidth}
-            height={textBgHeight}
-            backgroundColor={textBgColor}
-            opacity={textBgOpacity}
-            backgroundIsDisabled={isTextBgDisabled}
+            width={params.TBGWidth}
+            height
+            backgroundColor
+            opacity
+            backgroundIsDisabled
```
Comment (Hartha-aloufi) [inline src/pages/Main/muqtatif/Muqtatif.jsx]: what is this?

File: src/pages/Main/muqtatif/Muqtatif.jsx [hunk 7]
```diff
@@ -107,129 +58,25 @@ const Muqtatif = (props) => {
...
             dropShadow={
               !isTextBgDropShadowDisabled
-                ? `drop-Shadow(3px ${textBgDropShadowOffset}px ${textBgDropShadowBlur}px rgba(0, 0, 0, 0.45))`
+                ? `drop-Shadow(3px ${0}px ${0}px rgba(0, 0, 0, 0.45))`
```
Comment (Hartha-aloufi) [inline src/pages/Main/muqtatif/Muqtatif.jsx]: move this to css file since it is not a dynamic value

File: src/pages/Main/muqtatif/Muqtatif.jsx [hunk 8]
```diff
@@ -107,129 +58,25 @@ const Muqtatif = (props) => {
...
                 : "none"
             }
-            fontSize={textFontSize}
-            textColor={textColor}
-            fontWeight={textFontWeight}
-            fontFamily={fontFamily}
+            fontSize={fontSize}
+            textColor
+            fontWeight
+            fontFamily
```
Comment (Hartha-aloufi) [inline src/pages/Main/muqtatif/Muqtatif.jsx]: what is this?

File: src/pages/Main/muqtatif/muqHeader/settings/window/Window.jsx [hunk 9]
```diff
@@ -1,29 +1,75 @@
...
+} from "./WindowAtoms";
 
 const Window = (props) => {
-  const settingsCtx = useContext(SettingsContext);
+  const [isTextBgDisabled, setIsTextBgDisabled] = useState(false);
 
-  const [dropShadowIsChecked, setDropShadowIsChecked] = useState(false);
+  const [isSelected, setIsSelected] = useState("btn-b");
+  const [PresetSize, setPresetSize] = useAtom(aspectRatioAtom);
+  const presetSizeHandler = (e) => {
```
Comment (Hartha-aloufi) [inline src/pages/Main/muqtatif/muqHeader/settings/window/Window.jsx]: put this in usecallback

File: src/pages/Main/muqtatif/muqHeader/settings/Settings.jsx [hunk 10]
```diff
@@ -12,56 +12,55 @@ const Settings = (props) => {
...
+    //   }}
+    // >
+    <div className={classes.settingsContainer}>
+      <div className={classes.btnsSide}>
+        <RadioButtonsGroup
+          className={classes.radioBtns}
+          label={[
+            { id: "window", option: "Window", value: "window" },
+            { id: "editor", option: "Editor", value: "editor" },
+          ]}
```
Comment (Hartha-aloufi) [inline src/pages/Main/muqtatif/muqHeader/settings/Settings.jsx:22]: use Contants. wrap array with useMemo

File: src/pages/Main/muqtatif/muqHeader/settings/defaultValues.jsx [hunk 11]
```diff
@@ -0,0 +1,14 @@
+export const defaultValues = {
```
Comment (Hartha-aloufi) [inline src/pages/Main/muqtatif/muqHeader/settings/defaultValues.jsx]: use capital letters for constants naming

File: src/pages/Main/muqtatif/muqHeader/settings/window/Window.jsx [hunk 12]
```diff
@@ -1,29 +1,75 @@
...
+  dropShadowBlurAtom,
+} from "./WindowAtoms";
 
 const Window = (props) => {
-  const settingsCtx = useContext(SettingsContext);
+  const [isTextBgDisabled, setIsTextBgDisabled] = useState(false);
 
-  const [dropShadowIsChecked, setDropShadowIsChecked] = useState(false);
+  const [isSelected, setIsSelected] = useState("btn-b");
+  const [PresetSize, setPresetSize] = useAtom(aspectRatioAtom);
```
Comment (Hartha-aloufi) [inline src/pages/Main/muqtatif/muqHeader/settings/window/Window.jsx]: error in naming, use small letters for states

**Review by Hartha-aloufi — CHANGES_REQUESTED** [review]

File: src/pages/Main/muqtatif/Muqtatif.jsx (hunk 6 above)
Comment (orwa97) [reply src/pages/Main/muqtatif/Muqtatif.jsx]: props didnt give 'em values yet

File: src/pages/Main/muqtatif/Muqtatif.jsx (hunk 7 above)
Comment (orwa97) [reply src/pages/Main/muqtatif/Muqtatif.jsx]: they're dynamics but didnt pass the variables bcoz i didnt work on 'em yet

File: src/pages/Main/muqtatif/Muqtatif.jsx (hunk 8 above)
Comment (orwa97) [reply src/pages/Main/muqtatif/Muqtatif.jsx]: props i'll give 'em values later

File: src/pages/Main/muqtatif/muqHeader/settings/window/Window.jsx (hunk 9 above)
Comment (orwa97) [reply src/pages/Main/muqtatif/muqHeader/settings/window/Window.jsx]: wrap every Handler function with useCallback?!

File: src/pages/Main/muqtatif/muqHeader/settings/defaultValues.jsx (hunk 11 above)
Comment (orwa97) [reply src/pages/Main/muqtatif/muqHeader/settings/defaultValues.jsx]: u mean make just the object's name  capital letter like this DEFAULTVALUES? or change all the object's keys to capital letters?

Comment (Hartha-aloufi) [reply src/pages/Main/muqtatif/muqHeader/settings/defaultValues.jsx]: DEFAULT_VALUES

File: src/pages/Main/muqtatif/muqHeader/settings/editor/Editor.jsx [hunk 13]
```diff
@@ -11,16 +11,48 @@ import ColorButton from "../colorButton/ColorButton";
...
+  const [textColor, setTextColor] = useAtom(fontColorAtom);
+  const textColorHandler = (e) => {
+    setTextColor(e.hex);
+  };
+
+  const [textAlign, setTextAlign] = useAtom(textAlignAtom);
+  const [selectedTextAlign, setSelectedTextAlign] = useState("AR");
+  const textAlignHandler = (e) => {
+    setTextAlign(e.target.value);
+    setSelectedTextAlign(e.target.id);
```
Comment (Hartha-aloufi) [inline src/pages/Main/muqtatif/muqHeader/settings/editor/Editor.jsx]: الافضل تكون تعريفات الدوال وتعريفات الستيت منفصلة. عرف كل الستيت اولاً, ثم عرف كل الدوال

File: src/pages/Main/muqtatif/muqHeader/settings/window/Window.jsx (hunk 9 above)
Comment (Hartha-aloufi) [reply src/pages/Main/muqtatif/muqHeader/settings/window/Window.jsx]: المفروض هيك, لكن ان كانت الكمبوننت الي موديلها البروبس صغيرة, كبسة مثلا, فما في داعي

File: src/pages/Main/muqtatif/quoteArea/QuoteArea.jsx [hunk 14]
```diff
@@ -1,25 +1,60 @@
...
+    height: `${useAtomValue(textBgHeightAtom)}%`,
+    backgroundColor: useAtomValue(textBgColorAtom),
+    opacity: useAtomValue(textBgOpacityAtom),
+    fontSize: `${useAtomValue(fontSizeAtom)}rem`,
+    color: useAtomValue(fontColorAtom),
+    fontFamily: useAtomValue(fontFamilyAtom),
+    textAlign: useAtomValue(textAlignAtom),
+    lineHeight: useAtomValue(lineHeightAtom),
+    dropShadow: useAtomValue(dropShadowAtom) ? dropShadow : "none",
+  };
```
Comment (Hartha-aloufi) [inline src/pages/Main/muqtatif/quoteArea/QuoteArea.jsx:37]: useMemo

**Review by Hartha-aloufi — CHANGES_REQUESTED** [review]

File: src/pages/Main/muqtatif/quoteArea/QuoteArea.jsx (hunk 14 above)
Comment (orwa97) [reply src/pages/Main/muqtatif/quoteArea/QuoteArea.jsx:37]: useAtomValue is a hook  .. so, cant do that

**Comment by Hartha-aloufi** [comment]
حل الكونفليكت

**Comment by Hartha-aloufi** [comment]
حل الكونفليكت

**Pull request merged**